### PR TITLE
Switch to add_parsable_certificates

### PR DESCRIPTION
**Pull request by matthme**
Should address https://github.com/holochain/holochain/issues/4418

### CODE REVIEW

**Comment by matthme** [comment]
According to the documentation the [add](https://docs.rs/rustls/latest/rustls/struct.RootCertStore.html#method.add) method is apparently known to fail for some certificates so I switched to the [add_parsable_certificates](https://docs.rs/rustls/latest/rustls/struct.RootCertStore.html#method.add_parsable_certificates) method instead now.

File: crates/tx5-signal/Cargo.toml [hunk 1]
```diff
@@ -31,8 +31,6 @@ tokio-tungstenite = { workspace = true }
 tracing = { workspace = true }
 tx5-core = { workspace = true }
 url = { workspace = true }
-
-[target.'cfg(not(any(target_os = "windows", target_os = "linux", target_os = "macos")))'.dependencies]
```
Comment (ThetaSinner) [inline crates/tx5-signal/Cargo.toml:35]: This doesn't appear to be platform specific in any way, but it looks like it was deliberately only included for platforms that won't have a certificate store. Now that we're successfully using the certificate store on windows and this isn't being used on the platforms it's been included for, should this be reverted?

**Comment by ThetaSinner** [comment]
Tested on Windows where this was crashing for me, confirmed that this solves the problem and I can still make connections

File: crates/tx5-signal/Cargo.toml (hunk 1 above)
Comment (neonphog) [reply crates/tx5-signal/Cargo.toml:35]: Yes, this should be reverted. `rustls_native_certs` doesn't currently work on android or ios. (although it'd be nice if it did : )

Comment (matthme) [reply crates/tx5-signal/Cargo.toml:35]: Oh yes, sorry. I added this with the initial approach where the certs would be bundled.

**Review by neonphog — APPROVED** [review]
:+1:

**Review by ThetaSinner — APPROVED** [review]

**Pull request merged**